CodeEdit/AppDelegate.swift

```diff
@@ -234,10 +234,9 @@ final class AppDelegate: NSObject, NSApplicationDelegate, ObservableObject {
 
 /// Setup all the services into a ServiceContainer for the application to use.
 private func setupServiceContainer() {
-    // Example for how services will be instantiated
-//    ServiceContainer.register(
-//        PasteboardService()
-//    )
+    ServiceContainer.register(
```

Comment:
At what context level are we creating/injecting this object?
 - Application level?
 - Workspace level? (seems the most logical to me)
 - Editor level?
 - Another level?

Reply:
That's on the Application level, which is a problem for some objects that should be registered at the Workspace/Editor level. It's fine to leave it for now. I'll add functionality to `ServiceContainer` to initialise classes at the Workspace level in the upcoming Activity Viewer PR... or at least I'll try 😅

Reply:
I couldn’t resolve this issue. We should consider either adding CDI or using Swift's Environment object.